edward/criticisms/evaluate.py

```diff
@@ -115,26 +116,43 @@ def evaluate(metrics, data, n_samples=500, output_key=None):
   if metrics != ['log_lik'] and metrics != ['log_likelihood']:
     binary_discrete = (Bernoulli, Binomial)
     categorical_discrete = (Categorical, Multinomial, OneHotCategorical)
+    total_count = sess.run(getattr(output_key, 'total_count', tf.constant(1.)))
     if isinstance(output_key, binary_discrete + categorical_discrete):
       # Average over realizations of their probabilities, then predict
       # via argmax over probabilities.
       probs = [sess.run(output_key.probs, feed_dict) for _ in range(n_samples)]
-      probs = tf.add_n(probs) / tf.cast(n_samples, probs[0].dtype)
+      probs = np.sum(probs, axis=0) / n_samples
       if isinstance(output_key, binary_discrete):
         # make random prediction whenever probs is exactly 0.5
         random = tf.random_uniform(shape=tf.shape(probs))
         y_pred = tf.round(tf.where(tf.equal(0.5, probs), random, probs))
       else:
-        y_pred = tf.argmax(probs, len(probs.shape) - 1)
+        if total_count > 1:
+          mode = compute_multinomial_mode(probs, total_count, seed)
+          if len(output_key.sample_shape):
```

Comment:
Can you explain the use case for this part or provide an example? I'm not sure I followed this logic.